internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -91,6 +97,13 @@ const (
 	FunctionContextKeyEnvironment = "apiextensions.crossplane.io/environment"
 )
 
+const (
+	// MaxRequirementsIterations is the maximum number of times a Function should be called,
+	// limiting the number of times it can request for extra resources, capped for
+	// safety. We might allow Composition authors customize this value in the future.
```

Comment:
```suggestion
	// safety.
```

Nit: I'd remove this from the type documentation. I think it's fine left unsaid, but you could add a distinct TODO comment nearby if you prefer.

Reply:
removed